pkg/deploy/internal/classic/validation.go

```diff
@@ -51,6 +61,24 @@ func (v *Validator) Validate(c config.Config) error {
 	}
 
 	for _, c2 := range v.uniqueNames[c.Environment][a.Api] {
+
+		// if the configs have a scope and it is different then the configs are unique
+		if theAPI.HasParent() {
+			s1, err := config.GetScopeParameterForConfig(c)
+			if err != nil {
+				return err
```

Comment:
Do we actually need to treat not having a scope defined as an error here? Shouldn't loading already fail and ensure that a scope is set for APIs that need one? 

Then we could simplify this to accessing the scope parameters as `c.Parameters[ScopeParameter]` and comparing them? If one accidentally happens to be nil, then they are still different for the sake of validation

Reply:
No idea, I'm going to test this. But for the moment, I have put a marker in the error message so that I can see if it occurs

Reply:
OK, after some testing, it is possible to hit this error by having `scope: null` in the `config.yaml`

It is interesting that we get different behavior for `scope: ~`, `scope: {}` and `scope: null`.